File: entregavel/ordenacao.java
public class OrdenacaoBubbleSort {
    public static void main(String[] args) {
        // Nosso vetor desordenado
        int[] vetor = {5, 2, 9, 1, 5, 6};

        System.out.println("Vetor original:");
        imprimirVetor(vetor);

        // Aplica o algoritmo de ordenação
        bubbleSort(vetor);

        System.out.println("\nVetor ordenado:");
        imprimirVetor(vetor);
    }

    // Função que ordena o vetor
    public static void bubbleSort(int[] vetor) {
        int n = vetor.length;
        
        // O laço externo controla quantas vezes vamos passar pelo vetor
        for (int i = 0; i < n - 1; i++) {
            // O laço interno faz as comparações par a par
            for (int j = 0; j < n - 1 - i; j++) {
                // Se o elemento atual for maior que o próximo, eles trocam de lugar
                if (vetor[j] > vetor[j + 1]) {
                    int temp = vetor[j];
                    vetor[j] = vetor[j + 1];
                    vetor[j + 1] = temp;
                }
            }
        }
    }

    // Função auxiliar apenas para exibir o vetor no terminal
    public static void imprimirVetor(int[] vetor) {
        for (int numero : vetor) {
            System.out.print(numero + " ");
        }
        System.out.println();
    }
}

File: entregavel/ordenacao.js
// Função que ordena o array
function bubbleSort(vetor) {
    let n = vetor.length;
    
    for (let i = 0; i < n - 1; i++) {
        for (let j = 0; j < n - 1 - i; j++) {
            // Se o elemento atual for maior que o próximo, troca de lugar
            if (vetor[j] > vetor[j + 1]) {
                let temp = vetor[j];
                vetor[j] = vetor[j + 1];
                vetor[j + 1] = temp;
            }
        }
    }
    return vetor;
}

// Testando o código
let dados = [5, 2, 9, 1, 5, 6];

console.log("Vetor original:", dados);
console.log("Vetor ordenado:", bubbleSort(dados));
